sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceivedMessage.java

```diff
@@ -554,7 +579,11 @@ void setLockedUntil(OffsetDateTime lockedUntil) {
      * @param messageId to be set.
      */
     void setMessageId(String messageId) {
-        amqpAnnotatedMessage.getProperties().setMessageId(messageId);
+        AmqpMessageId id = null;
+        if (messageId != null) {
+            id = new AmqpMessageId(messageId);
+        }
+        amqpAnnotatedMessage.getProperties().setMessageId(id);
```

Comment:
Will service bus never support the other AmqpMessageId formats and will only ever support String?

Reply:
My understanding is that service bus will expose `String` only but advance user who want to stay in AMQP land, they can get AmqpAnnotatedMesage and set MessageId in other format whenever we provide support for it.
@ramya-rao-a  Is this assumption correct ?

Reply:
@JoshLove-msft has the details on this from the conversation with Clemens

Reply:
If there is a future scenario where these other types are relevant, we should have a plan for how they will be enabled so that we can be sure it will be possible in a non-breaking fashion.

Reply:
The service already supports all formats. We didn't enable it yet in .NET because the AMQP lib doesn't allow for getting back the data in its underlying format - it only allows you to get it as a string. 

The way we will support it in .NET would be to have different ctors for each of the additional formats.
We will also have TryGetX methods where users can attempt to get back the data for each format, and an enum Format Property that has the format, similar to how the AmqpMessageBody/BodyType works.

Reply:
When you speak to how you will do this in .NET, are you speaking to how you would implement ServiceBusReceivedMessage? Right now we are looking at methods in Java such as `setMessageId` here, and it sounds like your proposed way forward would require a number of setter overloads for each type (or else, support for setting the `AmqpMessageId` directly)

Reply:
No this would be for setting it directly on the AmqpAnnotatedMessage. We would not expose this on the ServiceBus API directly.

Reply:
So the Service Bus APIs will continue to be String-only forever? @hemanttanwar Can you provide sample code on how this would be done (either today or in the future) for Java using the approach Josh shares above?